.github/workflows/main.yaml

```diff
@@ -0,0 +1,45 @@
+name: Run tests with Tox
+
+on: [push]
+
+jobs:
+  unit-tests:
+    name: Lint, Unit Tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: [3.6, 3.7, 3.8, 3.9]
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Install Dependencies
+        run: pip install tox
+      - name: Run lint
+        run: tox -vve lint
+      - name: Run test
+        run: tox -e unit
+
+  integration-tests:
+    name: Integration test with Microstack
+    runs-on: self-hosted
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@master
```

Comment:
This will fail since the actions-operator master branch no longer exists. It needs to updated to use `@main` instead.